src/plugins/intel_npu/src/al/include/intel_npu/npu_mlir_runtime.hpp

```diff
@@ -0,0 +1,156 @@
+#    pragma once
+#endif
+
+#include "ze_api.h"
+#include "ze_graph_ext.h"
+
+#if defined(__cplusplus)
+#    include <cstdint>
+#    include <cstdlib>
+#else
+#    include <stdint.h>
+#    include <stdlib.h>
+#endif
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Supported versions
+///
+/// @details
+///     - Graph extension versions contain major and minor attributes, use
+///       ::NPU_MLIR_RUNTIME_MAJOR_VERSION and ::NPU_MLIR_RUNTIME_MINOR_VERSION
+typedef enum _npu_mlir_runtime_version_t {
+    NPU_MLIR_RUNTIME_VERSION_1_0 = ZE_MAKE_VERSION(1, 0),             ///< version 1.0
+    NPU_MLIR_RUNTIME_VERSION_CURRENT = NPU_MLIR_RUNTIME_VERSION_1_0,  ///< latest known version
+    NPU_MLIR_RUNTIME_VERSION_FORCE_UINT32 = 0x7fffffff,
+} npu_mlir_runtime_version_t;
+
+///////////////////////////////////////////////////////////////////////////////
+#ifndef NPU_MLIR_RUNTIME_APICALL
+#    if defined(_WIN32)
+/// @brief Calling convention for all API functions
+#        define NPU_MLIR_RUNTIME_APICALL __cdecl
+#    else
+#        define NPU_MLIR_RUNTIME_APICALL
+#    endif  // defined(_WIN32)
+#endif      // NPU_MLIR_RUNTIME_APICALL
+
+///////////////////////////////////////////////////////////////////////////////
+#ifndef NPU_MLIR_RUNTIME_APIEXPORT
+#    if defined(_WIN32)
+/// @brief Windows-specific dllexport storage-class attribute
+#        define NPU_MLIR_RUNTIME_APIEXPORT __declspec(dllexport)
+#    else
+#        define NPU_MLIR_RUNTIME_APIEXPORT
+#    endif  // defined(_WIN32)
+#endif      // NPU_MLIR_RUNTIME_APIEXPORT
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief NPU MLIR runtime handle
+typedef struct _npu_mlir_runtime_handle_t* npu_mlir_runtime_handle_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Defined Return/Error codes
+typedef enum _npu_mlir_runtime_result_t {
+    NPU_MLIR_RUNTIME_RESULT_SUCCESS = 0,
+    NPU_MLIR_RUNTIME_RESULT_ERROR_INVALID_NULL_POINTER = 0x80000001,
+    NPU_MLIR_RUNTIME_RESULT_ERROR_UNKNOWN = 0x8ffffffe,
+    NPU_MLIR_RUNTIME_RESULT_FORCE_UINT32 = 0x8fffffff,
+} npu_mlir_runtime_result_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Blob descriptor
+typedef struct _npu_mlir_runtime_blob_desc_t {
+    size_t inputSize;       ///< [in] Size of input buffer in bytes
+    const uint8_t* pInput;  ///< [in] Pointer to input buffer
+} npu_mlir_runtime_blob_desc_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Runtime properties
+typedef struct _npu_mlir_runtime_properties_t {
+    uint32_t numOfSubGraphs;
+    uint32_t numOfGraphArgs;
+} npu_mlir_runtime_properties_t;
+
+typedef struct _npu_mlir_runtime_mem_ref_t {
+    const void* basePtr;
+    const void* data;
+    int64_t offset;
+    int64_t sizes[4];
```

Comment:
Just a reminder for me to file a ticket to extend to 5d and support arbitrary rank for MLIR runtime

Reply:
The ticket is E#191301